pkg/server/endpoints/endpoints.go

```diff
@@ -46,6 +46,10 @@ const (
 	// This is the default amount of time between two reloads of the in-memory
 	// entry cache.
 	defaultCacheReloadInterval = 5 * time.Second
+
+	// This is the default amount of time between two reloads of the in-memory
+	// entry cache.
+	defaultCachePruneEventsInterval = 20 * time.Second
```

Comment:
+1, I can say from experience that the entry cache sync itself can be over 20 seconds with larger deployments under high load today. I would expect the cache hydration to be much, much faster with this approach, but there are some reliability considerations here as well.

For example, if the server in-memory entry cache sync routine fails consecutive times (e.g. temporary network partition between server instance and database), it could cause some server instances to miss events when the server is finally able to re-sync with the database.

I think this prune interval value should be something conservative initially, like a few hours. That gives an operator some time to mitigate any incidents that may be causing the entry cache sync to fail for some period of time. Ideally an operator shouldn't need to restart the server to bring the cache back to a healthy state.

During the initial usage of this feature before it's mainstreamed, keeping the events around a while also may be useful to debug state in case there are any anomalies/bugs found where events may have gotten missed. I personally don't see any harm in keeping events around for a while to start out with, as the event data is quite small in general.

We can also probably validate later on through some testing whether bumping this to something like 6-12 hours has any noticeable effect on entry cache sync performance. In any case, it should still be much faster than the current approach.

Reply:
20 seconds was just a test value. Its now updated to 12 hours. Can fine tune that further.